src/python/pants/backend/project_info/paths.py

```diff
@@ -0,0 +1,137 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import json
+from collections import deque
+from operator import attrgetter
+from typing import Dict, Generator, cast
+
+from pants.engine.addresses import Address, Addresses, UnparsedAddressInputs
+from pants.engine.console import Console
+from pants.engine.goal import Goal, GoalSubsystem, LineOriented
+from pants.engine.rules import Get, MultiGet, collect_rules, goal_rule
+from pants.engine.target import Dependencies as DependenciesField
+from pants.engine.target import (
+    DependenciesRequest,
+    Targets,
+    TransitiveTargets,
+    TransitiveTargetsRequest,
+)
+
+
+class PathsSubsystem(LineOriented, GoalSubsystem):
+    name = "paths"
+    help = "List the paths between two addresses."
+
+    @classmethod
+    def register_options(cls, register):
+        super().register_options(register)
+        register(
+            "--path-from",
```

Comment:
The options are scoped already by the subsystem, so can be `--from` and `--to` here.

Reply:
You're right, but I have to change the param names because `from` is a reserved keyword in python.
`target`/`root` and `dependency`?